tests/test_coverity.py

```diff
@@ -1,126 +1,236 @@
+        return coverity_service
+
+    def test_session_by_stream_validation(self):
+        self.fake_stream = "test_stream"
+        coverity_service = self.initialize_coverity_service(login=False)
+        with requests_mock.mock() as mocker:
+            mocker.get(self.stream_url, json={})
+            # Login to Coverity
+            coverity_service.login("user", "password")
+            coverity_service.validate_stream(self.fake_stream)
+            stream_request = mocker.last_request
+            assert stream_request.headers["Authorization"] == requests.auth._basic_auth_str("user", "password")
+
+    @patch("mlx.coverity_services.requests")
+    def test_stream_validation(self, mock_requests):
+        mock_requests.return_value = MagicMock(spec=requests)
+
+        self.fake_stream = "test_stream"
+        # Get the base url
+        coverity_service = CoverityDefectService("scan.coverity.com/")
+        # Login to Coverity
+        coverity_service.login("user", "password")
+        with patch.object(CoverityDefectService, "_request") as mock_method:
+            # Validate stream name
+            coverity_service.validate_stream(self.fake_stream)
+            mock_method.assert_called_once()
+            mock_method.assert_called_with("https://scan.coverity.com/api/v2/streams/test_stream")
+
+    def test_retrieve_columns(self):
+        with open(f"{TEST_FOLDER}/columns_keys.json", "r") as content:
+            column_keys = json.loads(content.read())
+        self.fake_stream = "test_stream"
+        # initialize what needed for the REST API
+        coverity_service = self.initialize_coverity_service(login=True)
+        with requests_mock.mock() as mocker:
+            mocker.get(self.column_keys_url, json=column_keys)
+            coverity_service.retrieve_column_keys()
+            assert mocker.call_count == 1
+            mock_request = mocker.last_request
+            assert mock_request.method == "GET"
+            assert mock_request.url == self.column_keys_url
+            assert mock_request.verify
+            assert coverity_service.columns["Issue Kind"] == "displayIssueKind"
+            assert coverity_service.columns["CID"] == "cid"
+
+    def test_retrieve_checkers(self):
+        self.fake_stream = "test_stream"
+        self.fake_checkers = {
+            "checkerAttribute": {"name": "checker", "displayName": "Checker"},
+            "checkerAttributedata": [
+                {"key": "MISRA", "value": "MISRA"},
+                {"key": "CHECKER", "value": "CHECKER"}
+            ],
+        }
+        # initialize what needed for the REST API
+        coverity_service = self.initialize_coverity_service(login=True)
+
+        with requests_mock.mock() as mocker:
+            mocker.get(self.checkers_url, json=self.fake_checkers)
+            coverity_service.retrieve_checkers()
+            history = mocker.request_history
+            assert mocker.call_count == 1
+            assert history[0].method == "GET"
```

Comment:
I think you should use `mock_request = mocker.last_request` for readability and consistency with `test_retrieve_columns`.

Reply:
Resolved in 4e8c8580dafa8be2376c658e725f4ea89da7cd5d